pydantic/_internal/_typing_extra.py

```diff
@@ -308,7 +308,7 @@ def eval_type_backport(
 
         # If it is a `TypeError` and value isn't a `ForwardRef`, it would have failed during annotation definition.
         # Thus we assert here for type checking purposes:
-        assert isinstance(value, typing.ForwardRef)
+        assert isinstance(value, typing.ForwardRef), (value, e)
```

Comment:
```suggestion
        assert isinstance(value, typing.ForwardRef)
```
I need to come back to this for https://github.com/pydantic/pydantic/issues/10565